Page on installing plugins to Neo4j (#863)
neo4j/docs-operations: modules/ROOT/pages/configuration/plugins.adoc

[[plugin-settings]]

> = Plugin settings

Author: I changed this to only "plugins". Do you think it's better to have it as configure plugins instead?

[[plugins]]
= Plugins
:description: This page describes how to load plugins to a Neo4j deployment.

> This page describes how to load plugins to a Neo4j deployment.

Reviewer: I might want to add here some intro about the plugins as this would be the main page in the manual about plugins. We should mention what they are and how they are meant to be used, and that some of them require activation key.

The following plugins are supported:

.Supported Neo4j plugins
[options="header",cols="d,m,b,a"]
|===
|Name |Key  | Licence required | Further information

| APOC
| `apoc`
| {cross-mark}
| https://neo4j.com/docs/apoc/current/[APOC user guide]

| Bloom
| `bloom`
| {check-mark}
| link:{neo4j-docs-base-uri}/bloom-user-guide[Neo4j Bloom]

| Streams
| `streams`
| {cross-mark}
| link:{neo4j-docs-base-uri}/kafka-streams[Neo4j Streaming Data Integrations User Guide]

| Graph Data Science
| `graph-data-science`
| {cross-mark}
| link:{neo4j-docs-base-uri}/graph-data-science[Graph Data Science]

| Neo Semantics
| `n10s`
| {cross-mark}
| https://neo4j.com/labs/nsmtx-rdf/[Neo Semantics]
|===

[NOTE]
====

> Refer to these other pages on plugins in case you are using link:https://neo4j.com/docs/desktop-manual/current/operations/install-plugin/[Neo4j Desktop], xref:docker/operations.adoc#docker-neo4j-plugins[Docker], or xref:/kubernetes/plugins/[Kubernetes]. 

Reviewer: I think it is a good practice to also give the names of the pages we are referring to in case the links are broken. For example,
```suggestion
For more information on using plugins in a different Neo4j setup, see link:{neo4j-docs-base-uri}/desktop-manual/current/operations/install-plugin/[Neo4j Desktop -> Install a plugin], xref:docker/operations.adoc#docker-neo4j-plugins[Docker -> Configure Neo4j plugins],  xref:/kubernetes/plugins/[Kubernetes], and link:{neo4j-docs-base-uri}/java-reference/{page-version}/extending-neo4j/project-setup/#_build_dependencies[Java-Reference -> Setting up a plugin project]. 
```

====

> == Loading plugins 

Reviewer: ```suggestion
== Install and configure plugins 
```

Here are the steps to load the plugins:

. Go to `<NEO4J_HOME>/products` and copy or move the chosen plugins (.jar files) into the `<NEO4J_HOME>/plugins` directory.
See the table in xref:configuration/file-locations/[File locations] for more information.

> . *Optional*: Some plugins, such as Bloom, require licencing and configuring to become available to the user within Neo4j.

Reviewer: I think all of them require conifguration but some also require an activation key.

> . *Optional*: Some plugins, such as Bloom, require licencing and configuring to become available to the user within Neo4j.

Reviewer: ```suggestion
. *Optional*: Some plugins, such as Bloom, require licensing and configuring to become available to the user within Neo4j.
```

Reviewer: American English 😄

Author: This was a tricky one! I always think the word with "s" is British 😂

Refer to link:https://neo4j.com/docs/bloom-user-guide/current/bloom-installation/installation-activation/[Bloom's installation and activation] for more details.

[NOTE]
====

> All installed plugins will automatically be loaded every time Neo4j is opened.

Reviewer: This I don't know if it is true. Maybe, you could ask someone from kernel?

Author: This was something that Stu added to the card, but I will ask someone from Kernel!

Reviewer 2: The plugins-folder is scanned at DBMS startup, and all JARs are loaded into the application.

The DBMS is started when the Neo4j Desktop application is started - but for a server setup the DBMS service should be restarted.

| `apoc`
| {cross-mark}
| https://neo4j.com/docs/apoc/current/[APOC user guide]

| Bloom
| `bloom`
| {check-mark}
| link:{neo4j-docs-base-uri}/bloom-user-guide[Neo4j Bloom]

| Streams
| `streams`
| {cross-mark}
| link:{neo4j-docs-base-uri}/kafka-streams[Neo4j Streaming Data Integrations User Guide]

| Graph Data Science
| `graph-data-science`
| {cross-mark}
| link:{neo4j-docs-base-uri}/graph-data-science[Graph Data Science]

| Neo Semantics
| `n10s`
| {cross-mark}
| https://neo4j.com/labs/nsmtx-rdf/[Neo Semantics]

| Ops Manager
| `ops-manager`
| {cross-mark}
| link:{neo4j-docs-base-uri}/ops-manager/[Neo4j Ops Manager]
|===

[NOTE]
====
For more information on using plugins in a different Neo4j setup, see link:{neo4j-docs-base-uri}/desktop-manual/current/operations/install-plugin/[Neo4j Desktop -> Install a plugin], xref:docker/operations.adoc#docker-neo4j-plugins[Docker -> Configure Neo4j plugins],  xref:/kubernetes/plugins/[Kubernetes], and link:{neo4j-docs-base-uri}/java-reference/{page-version}/extending-neo4j/project-setup/#_build_dependencies[Java-Reference -> Setting up a plugin project]. 
====

== Install and configure plugins 

Here are the steps to load the plugins:

> . Download the plugin _.jar file_ and move it to the _<NEO4J_HOME>/plugins_ directory.

Reviewer 2: **Consider** advising the users to first look in `<NEO4J_HOME>/products` and `<NEO4J_HOME>/labs`, and otherwise download from the internet.

I believe especially `apoc-core` is quite fussy about which versions are supported for which DBMS version, so it is probably easiest to use the bundled version to get it to work.

[[plugins]]
= Plugins
:description: This page describes how to load plugins to a Neo4j deployment.

> For each DBMS, active or not, you have the option to install a plugin library which will extend the capabilities of your database.

Reviewer: ```suggestion
Neo4j distributions come bundled with a range of pre-installed products, such as Graph Data Science library, Bloom, and Ops Manager, which can be used to extend the Neo4j capabilities.
The JAR files for these products are located in the _product_ folder and can be installed as plugins.

If you want to use your own plugins, ensure that you add them to the designated _plugins_ directory.
This directory serves as the central location where Neo4j looks for and loads the plugins at startup. 

[NOTE]
====
Bloom and GDS Enterprise require a license activation key to become available to the user within Neo4j.
Reach out to your Neo4j account representative or request a representative to link:https://neo4j.com/contact-us/#sales-inquiry[contact you].
====
```

> It must, however, be a local database since this feature is not available for remote databases.

Author: I found this in the [Desktop manual](https://neo4j.com/docs/desktop-manual/current/operations/install-plugin/), but maybe it's not valid here

> It must, however, be a local database since this feature is not available for remote databases.

Reviewer: I think this is Desktop specific.
